lib/ethereumex/counter.ex

```diff
@@ -19,11 +19,29 @@ defmodule Ethereumex.Counter do
 
   @spec increment(atom()) :: integer()
   def increment(key) do
-    :ets.update_counter(@tab, key, {2, 1}, {key, 0})
+    do_increment(Application.get_env(:ethereumex, :id_lock), key)
   end
 
   @spec increment(atom(), integer()) :: integer()
   def increment(key, count) do
+    do_increment(Application.get_env(:ethereumex, :id_lock), key, count)
```

Comment:
instead of using a lock, I suggest making changes to always start the counter from zero. IIUC Because json rpc uses request ids for batch request